bayesian/utils.py

```diff
@@ -420,6 +472,44 @@ def get_link(self):
         return "https://snyk.io/vuln/{}:{}".format(
                 snyk_ecosystem[self.ecosystem], quote(self.package))
 
+    def vendor_recommendation(self, clubbed_data, package, version):
+        """Recommendation function which adds Recommendation Text in Response."""
+        # Template Dict for recommendation
+        reco = {
+            'recommendation': {
+                'component-analyses': {},
+            },
+            'snyk_package_link': self.get_link()
+        }
+        message = '{vulnerable} vulnerabilities within this dependency, ' \
+                  '{exploitable} with exploitable vulnerabilities'
+
+        if clubbed_data:
+            gremlin_resp = SnykComponentAnalysisResponse(package, version, clubbed_data)
+
+            reco['data'] = gremlin_resp.results
+
+            if not gremlin_resp.has_cves():
+                # If there is No CVE, we have No Recommendations for you.
+                reco['recommendation'] = {}
+                return {"result": reco}
+
+            recommendation_dict = reco['recommendation']
```

Comment:
I would suggest you use .get instead of directly accessing values. If some field is missing, you will get an exception

Reply:
It will never be a case. 
`reco['recommendation']` is picked up from static template